CMakeLists.txt

```diff
@@ -18,6 +18,20 @@ set(CMAKE_ALLOW_LOOSE_LOOP_CONSTRUCTS true)
 # use our own version of FindBoost.cmake and other Find* scripts
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
 
+# function to remove a flag from a variable
+function(RemoveFlag VAR SCOPE FLAG)
+	MESSAGE("Removing ${FLAG} flag from ${VAR}")
+	separate_arguments(${VAR})
+	list(REMOVE_ITEM ${VAR} ${FLAG})
+	string(REPLACE ";" " " ${VAR} "${${VAR}}")
+	
+	if("${SCOPE}" STREQUAL "CACHE")
+		set(${VAR} "${${VAR}}" CACHE STRING "" FORCE)
```

Comment:
Likewise, here I'm not sure if the indirection is needed.

In addition, I'd like if you provided the `docstring` parameter. For these parameters, something like

> Default C++ flags for RelWithDebInfo
> Defaut C flags for RelWithDebInfo
> Default C++ flags for MinSizeRel
> Default C flags for MinSizeRel

Reply:
I could say "Removed ${FLAG}" in the docstring, but to have more would require either passing in more information as arguments or attempting to parse the variable name.  C vs C++ flags and the build being used should be fairly obvious from the variable name itself though, if/when someone does open up the CMakeCache.

Reply:
Passing the docstring as an argument is what I'd prefer.

And the docstring is supposed to mean what the variable **is**, not what we have done to it.